### ISSUE TITLE
[FEATURE]: Improve pagination options: hide excess selectors, allow an "All" option, and able to dynamically set paginationPerPage

### ISSUE THREAD

**Issue by cliffordp** [jbetancur/react-data-table-component#530]
## Is your feature request related to a problem? Please describe

1)
When pagination options exceed the total record count, it clutters. Why have 50, 100, and 1,000 show if there are only 22 records?
![image](https://user-images.githubusercontent.com/1812179/78912198-ebc6d100-7a4c-11ea-985d-509696f41d85.png)

2)
I'd like an **"All"** option, [like jQuery DataTables does](https://datatables.net/reference/option/lengthMenu) (using `-1`):

I don't know if I'll have 40 records or 4,000 but I want the user to be able to View All if they choose.

![image](https://user-images.githubusercontent.com/1812179/78917141-19fbdf00-7a54-11ea-970b-b15e0c0b75fa.png)

![image](https://user-images.githubusercontent.com/1812179/78916914-c9848180-7a53-11ea-8db0-ad042da89e2a.png)

### Custom code I tried:

```js
const paginationOpts = [ 5, 10, 20, 50, 100 ];

/**
 * Get the Per Page options, with the max equaling the length of data. Sorted numerically, ASC.
 *
 * @returns {number[]}
 */
const getPaginationRowsPerPage = () => {
	let arr = paginationOpts;

	// TODO: could x10 if data length is over 1,000

	if ( data ) {
		arr = arr.filter( ( item ) => item < data.length );
		arr.push( data.length );
	}

	arr.sort( ( a, b ) => a - b );

	return arr;
};

/**
 * Get the default Per Page value, selecting the lower of 20 or the length of the data.
 *
 * @returns {number}
 */
const getPaginationPerPage = () => {
	if ( ! data ) {
		return 20;
	}

	return paginationOpts.reverse().find( ( item ) => item < data.length );
};

<DataTable
	columns={dataTableColumns}
	data={data}
	persistTableHead
	progressPending={!data}
	progressComponent={<Placeholder><Spinner /></Placeholder>}
	selectableRows
	striped
	pagination
	paginationPerPage={getPaginationPerPage()} // **!!!DOES NOT WORK!!!**
	paginationRowsPerPageOptions={getPaginationRowsPerPage()} // DOES WORK :)
/>
```

## Describe the solution you'd like

Basically, I don't want to have to write all this code, plus I want the last option to be "All" text, not just an out-of-sync number (like `[ 5, 10, 20, 22 ]`), and the dynamic `paginationPerPage` doesn't work -- wish it did out of the box.

**Comment by jbetancur** [comment]
Here you go! Once I release 6.5.0 sometime this weekend this feature will be available.

To make it work just specify the following:

```js
    paginationComponentOptions={{
      selectAllRowsItem: true,
    }}
```

You can also see an example here:
https://jbetancur.github.io/react-data-table-component/?path=/story/pagination--options


Note that `onChangeRowsPerPage`, if you are using it, will return the data.length and not -1 as you are suggesting as this is what is required when using service side pagination.

**Comment by cliffordp** [comment]
Wow, awesome. tyvm! I see it solves my item#2.

Any thoughts about my item#1 getting built-in?

And how'd things jump from 6.5.0 to 6.7.1 so quickly? no 6.6.x?

**Comment by devosaini** [comment]
I had a question regarding the use of the 'All' option being implemented. Is there anyway to show all rows by default?

**Comment by cliffordp** [comment]
@jbetancur ☝

**Comment by devosaini** [comment]
I hope this could be possible, as it would then allow for loading results as the page/table was scrolled. This would be great, hope to hear back from you soon @jbetancur :)!
